File: ruby/soap/savon/get_flights_enroute.rb
#!/usr/bin/env ruby

require 'rubygems'
require 'savon'

SOAP_URL = 'https://flightxml.flightaware.com/soap/FlightXML3/wsdl'
USERNAME = 'YOUR_USERNAME'
API_KEY = 'YOUR_API_KEY'

client = Savon.client do
	wsdl SOAP_URL
	basic_auth [USERNAME, API_KEY]
	convert_request_keys_to :none
end

message = {
	:airport_code => 'KJFK',
	:include_ex_data => 0,
	:type => 'enroute',
	:howMany => 10
}

response = client.call(:airport_boards, message: message)

enroute = response.body[:airport_boards_results][:airport_boards_result][:enroute]

puts "Aircraft enroute to KJFK:"

enroute[:flights].each do |flight|
	puts "#{flight[:ident]} (#{flight[:aircrafttype]})\t#{flight[:origin][:airport_name]} (#{flight[:origin][:code]})"
end

File: python/soap/zeep/get_flights_enroute.py
#!/usr/bin/env python

from requests import Session
from requests.auth import HTTPBasicAuth
from zeep import Client
from zeep.transports import Transport

username = 'YOUR_USERNAME'
apiKey = 'YOUR_API_KEY'
wsdlFile = 'https://flightxml.flightaware.com/soap/FlightXML3/wsdl'

session = Session()
session.auth = HTTPBasicAuth(username, apiKey)
client = Client(wsdlFile, transport=Transport(session=session))
boards = client.service.AirportBoards('KHOU',0,'','enroute',10,0)

flights = boards['enroute']['flights']
print "Flight enroute to KHOU:"
for flight in flights:
	print "{} ({}) \t{} ({})".format(flight['ident'], flight['aircrafttype'],
					flight['origin']['airport_name'], flight['origin']['code'])